libs/core/libs/shared/src/core_ops.rs

```diff
@@ -128,15 +132,12 @@ where
             }
         }
 
-        for item in &mut files {
-            if let Some(new_parent) = parent_substitutions.get(&item.parent) {
-                item.parent = *new_parent;
-            }
-        }
-
         Ok(files)
     }
 
+    fn is_invisible_id(&mut self, id: Uuid) -> SharedResult<bool> {
+        Ok(self.calculate_deleted(&id)? || self.in_pending_share(&id)? || self.link(&id)?.is_some())
```

Comment:
if you agree with what I've said above, then the criteria for invisibility is:

invisible == true if:
+ deleted
+ in pending share
+ it's a link

Reply:
yeah isn't that the conditions that the function counts:
- deleted `self.calculate_deleted(&id)? `
- in pending share `self.in_pending_share(&id)?`
- it's a link `self.link(&id)?.is_some()`